controller/apps.go

```diff
@@ -0,0 +1,443 @@
+package controller
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/fabric8-services/fabric8-wit/auth/authservice"
+
+	"github.com/fabric8-services/fabric8-wit/app"
+	"github.com/fabric8-services/fabric8-wit/auth"
+	"github.com/fabric8-services/fabric8-wit/configuration"
+	witerrors "github.com/fabric8-services/fabric8-wit/errors"
+	"github.com/fabric8-services/fabric8-wit/kubernetes"
+	"github.com/fabric8-services/fabric8-wit/log"
+	"github.com/goadesign/goa"
+	uuid "github.com/satori/go.uuid"
+)
+
+// AppsController implements the apps resource.
+type AppsController struct {
+	*goa.Controller
+	Config *configuration.Registry
+}
+
+// NewAppsController creates a apps controller.
+func NewAppsController(service *goa.Service, config *configuration.Registry) *AppsController {
+	return &AppsController{
+		Controller: service.NewController("AppsController"),
+		Config:     config,
+	}
+}
+
+func tostring(item interface{}) string {
+	bytes, _ := json.MarshalIndent(item, "", "  ")
+	return string(bytes)
+}
+
+func (c *AppsController) getAndCheckOSIOClient(ctx context.Context) *OSIOClient {
+
+	// TODO - grab the WIT URL from the incoming request if possible
+	witURL := "localhost"
+
+	// TODO - remove this debug hook before production
+	if os.Getenv("OSIO_WIT_URL") != "" {
+		witURL = os.Getenv("OSIO_WIT_URL")
+	}
+
+	oc := NewOSIOClient(ctx, witURL)
+
+	return oc
+}
+
+func (c *AppsController) getSpaceNameFromSpaceID(ctx context.Context, spaceID uuid.UUID) (*string, error) {
+	// TODO - add a cache in AppsController
+	// use WIT API to convert Space UUID to Space name
+	oc := c.getAndCheckOSIOClient(ctx)
```

Comment:
using `osio` as in `getNamespaceName()` below would be better and more consistent, IMO